src/ts/components/Controls.tsx

```diff
@@ -57,6 +62,43 @@ class Controls extends Component<IProps, IState> {
       ? "Sidebar__header-container"
       : "Sidebar__header-container Sidebar__header-container--is-closed";
 
+    const renderErrorMessage = () => {
+      const pluginState = this.state.pluginState;
+
+      if (!pluginState) {
+        return;
+      }
+
+      const hasAuthError = selectHasAuthError(pluginState);
+      const hasGeneralError = selectHasGeneralError(pluginState);
+      const hasErrors: boolean = hasAuthError || hasGeneralError;
+
+      let errorMessage: string = "";
```

Comment:
Because we know we've only got two kinds of error, could this read:
```
const errorMessage = hasAuthError ? "...auth error msg" : "normal error msg"
```
We'd need to refactor this anyway if we added additional error types.

(We'd need to return earlier if there weren't any errors, re: comment below, for this to work)